ui/DesignSystem/Component/Sidebar.svelte

```diff
@@ -54,97 +56,46 @@
       rgba(0, 0, 0, 0) 100%
     );
   }
-
-  .item {
-    width: var(--sidebar-width);
-    height: 32px;
-    margin-bottom: 16px;
-    position: relative;
-    display: flex;
-    justify-content: center;
-    align-items: center;
-    cursor: pointer;
-  }
-
-  .indicator:hover:before {
-    position: absolute;
-    content: "";
-    width: 4px;
-    height: 32px;
-    background-color: var(--color-foreground-level-5);
-    top: 0px;
-    left: 0px;
-    border-top-right-radius: 4px;
-    border-bottom-right-radius: 4px;
-  }
-
-  .indicator.active:before {
-    position: absolute;
-    content: "";
-    width: 4px;
-    height: 32px;
-    background-color: var(--color-primary);
-    top: 0px;
-    left: 0px;
-    border-top-right-radius: 4px;
-    border-bottom-right-radius: 4px;
-  }
-
-  .indicator :global(div:hover svg) {
-    fill: var(--color-primary);
-  }
-
-  .indicator.active :global(svg) {
-    fill: var(--color-primary);
-  }
 </style>
 
 <div class="wrapper" data-cy="sidebar">
   <div class="top">
     <Tooltip value={identity.metadata.handle}>
-      <div
-        class="item indicator"
-        data-cy="profile"
-        class:active={$activeRouteStore.type === "profile"}
-        on:click|stopPropagation={() =>
-          push({ type: "profile", activeTab: "projects" })}>
+      <SidebarItem
+        dataCy="profile"
+        indicator
+        active={$activeRouteStore.type === "profile"}
+        on:itemClick={() => push({ type: "profile", activeTab: "projects" })}>
         <Avatar
           size="regular"
           avatarFallback={identity.avatarFallback}
           variant="circle" />
-      </div>
+      </SidebarItem>
     </Tooltip>
   </div>
   <div class="bottom">
     <Tooltip value="Navigate to a project">
-      <div
-        class="item indicator"
-        data-cy="search"
-        on:click|stopPropagation={() => modal.toggle(ModalSearch)}>
+      <SidebarItem
+        dataCy="search"
+        on:itemClick={() => modal.toggle(ModalSearch)}>
         <Icon.MagnifyingGlass />
-      </div>
+      </SidebarItem>
     </Tooltip>
     {#if config.isDev}
-      <Tooltip value="Wallet">
-        <div
-          class="item indicator"
-          data-cy="wallet"
-          class:active={$activeRouteStore.type === "wallet"}
-          on:click|stopPropagation={() =>
-            push({ type: "wallet", activeTab: "transactions" })}>
-          <Icon.Wallet />
-        </div>
-      </Tooltip>
+      <WalletStatusIndicator
+        active={$activeRouteStore.type === "wallet"}
+        on:walletClick={() =>
+          push({ type: "wallet", activeTab: "transactions" })} />
```

Comment:
We’re trying to [avoid](https://github.com/radicle-dev/radicle-upstream/issues/1521) using events because we can’t type them properly. Instead, we simply use callback props
```suggestion
        onClick={() =>
          push({ type: "wallet", activeTab: "transactions" })} />
```

And in `WalletStatusIndicator`
```typescript
export let onClick: () => void
```

Reply:
https://github.com/radicle-dev/radicle-upstream/pull/1930/commits/3f87e87b845bdb63021212e7cffd4fe17d2adbca